scratch.py

```diff
@@ -115,80 +117,96 @@ def pack_string(string):
                     incident.pilot_total_hours,
                     incident.midair
                 ))
-            fh.write(data)
+
+            file.write(data)
         except Exception as err:
             print(err)
+
         return True
 
     def import_binary(self, filename):
-        print("4343")
-        NumbersStruct = struct.Struct("<Idi?")
-        def unpack_string(fh, eof_is_error=True):
-            print(eof_is_error)
+        print("Entering `import_binary`")
+
+        def unpack_string(file, eof_is_error=False):
+            print(f"Entering `unpack_string`... {eof_is_error= }")
             uint16 = struct.Struct("<H")
-            length_data = fh.read(uint16.size)
-            print("yes", length_data,87)
+            length_data = file.read(uint16.size)
+            print(f"{length_data= }")
 
             if not length_data:
                 if eof_is_error:
                     raise ValueError("missing or corrupt string size")
-                return 78
+                return None
 
             length = uint16.unpack(length_data)[0]
-            print(length, "2343")
-            if length == 0:
+            print(f"{length= }")
+            if not length:
                 return ""
-            data = fh.read(length)
-            print(data,"re")
-            print(data,"me",len(data),length,len(data) != length)
-            if not data or len(data) != length:
```

Comment:
**Given the way the data is written to the file**, if `length` > 0 (`length` has been previously checked), the check `if not data:` is unnecessary.